### PR TITLE
Role deletion from Roles Table

### LINKED ISSUE #2393: [UI] Add a delete action in roles, clients, secrets, hooks views
Feature initially proposed in https://github.com/taskcluster/taskcluster/issues/2386#issue-568468798.

**Describe the bug**
In order to delete something from (e.g., https://community-tc.services.mozilla.com/auth/roles), a user needs to click on the role to delete, then click/hover on the triple dot fab button on the lower right then click on "Delete". That's a lot of clicks that we can perhaps avoid by having the list of hooks show a delete icon beside the name.

**To Reproduce**

1. Go to https://community-tc.services.mozilla.com/auth/roles

**Expected behavior**
A delete icon is shown beside each role. Clicking on the delete icon should behave the same way as the one in the speed dial (triple dots). In other words, it should first show a confirmation dialog prior to performing the actual delete.

**Taskcluster version**
v24.3.0

**Additional context**
Once we have a good idea about the implementation, we'll probably need to do the same on the following views:
* https://community-tc.services.mozilla.com/auth/clients
* https://community-tc.services.mozilla.com/secrets
* https://community-tc.services.mozilla.com/hooks

### PR DESCRIPTION
**Pull request by mrsVelena**
Github Bug/Issue: Fixes #2393

Added a delete action to each role in the list of roles

.

### CODE REVIEW

File: ui/src/components/RolesTable/index.jsx [hunk 1]
```diff
@@ -28,10 +53,12 @@ export default class RolesTable extends Component {
     onPageChange: func.isRequired,
     /** A search term to refine the list of roles */
     searchTerm: string,
+    onDialogActionOpen: func,
```
Comment (djmitche) [inline ui/src/components/RolesTable/index.jsx:56]: This prop is not checked for null-ness before being used, so it should have `.isRequired` like `onPageChange` does.  That also means you can omit it from `defaultProps`, below.

**Review by djmitche — CHANGES_REQUESTED** [review]
This looks like it's got the necessary pieces in place.  But, it has some linting failures that are preventing netlify from deploying it.  If you can get that fixed up, I can take another look.

The `service-auth` failure is because your work is based on an older branch (from a week or so ago).  It's fine to just leave that failing, as you haven't modified that service, but a rebase onto the latest commit on `main` will fix it.

**Comment by petemoore** [comment]
It looks like there may be some test failures in https://community-tc.services.mozilla.com/tasks/LHu8K21YQN6LmHKMD46ZQA/runs/0/logs/public/logs/live.log but I'm not too sure what is wrong...

**Review by djmitche — APPROVED** [review]
This is fantastic -- nice work!

One minor note that doesn't matter here, but will for the next PR: you should usually make a "topic branch" to work on a PR like this, so that your `main` branch can be updated to new versions of the shared `main` without losing your changes.  The [GitHub Flow](https://guides.github.com/introduction/flow/) doc has more information on that.  Once this is merged, you'll want to do something like

```
git checkout main
git fetch https://github.com/taskcluster/taskcluster main
git reset --hard FETCH_HEAD
```
and that will reset your `main` branch to the same commit as the taskcluster repo's `main` branch. Then you can create a topic branch for your next project:
```
git checkout -b my-next-project
# (or whatever you want to call the branch)
```

**Pull request merged**